core/capabilities/ccip/ocrimpls/svm_contract_transmitter_factory.go

```diff
@@ -41,15 +39,24 @@ type SVMExecCallArgs struct {
 }
 
 // SVMContractTransmitterFactory implements the transmitter factory for SVM chains.
-type SVMContractTransmitterFactory struct{}
+type SVMContractTransmitterFactory struct {
+	extraDataCodec ccipcommon.ExtraDataCodec
+}
+
+// NewSVMContractTransmitterFactory returns a new SVMContractTransmitterFactory.
+func NewSVMContractTransmitterFactory(extraDataCodec ccipcommon.ExtraDataCodec) *SVMContractTransmitterFactory {
+	return &SVMContractTransmitterFactory{
+		extraDataCodec: extraDataCodec,
+	}
+}
```

Comment:
Is there a specific reason this type is public?

```suggestion
type contractTransmitterFactorySVM struct {
	extraDataCodec ccipcommon.ExtraDataCodec
}

// NewSVMContractTransmitterFactory returns a new SVMContractTransmitterFactory.
func NewSVMContractTransmitterFactory(extraDataCodec ccipcommon.ExtraDataCodec) oraclecreator.ContractTransmitterFactory {
	return &contractTransmitterFactorySVM{extraDataCodec}
}
```

I'd suggest only exposing the necessary types and APIs, keep internal by default - not everything public by default.

Reply:
NewSVMContractTransmitterFactory() in called in ccipsolana folder